domains/auth/Auth.API/API/Services/OidcProviders/SignaturGruppen.cs

```diff
@@ -1,30 +1,34 @@
+using System.Net;
+using System.Text;
 using System.Text.Json;
 using API.Configuration;
-using API.Helpers;
+using API.Controllers.dto;
+using API.Errors;
 using API.Models;
+using API.Utilities;
 using Microsoft.AspNetCore.Http.Extensions;
 using Microsoft.Extensions.Options;
 
 namespace API.Services.OidcProviders;
 
 public class SignaturGruppen : IOidcService
 {
-    private readonly IOidcService oidcService;
     private readonly AuthOptions authOptions;
     private readonly ILogger<SignaturGruppen> logger;
     private readonly HttpClient httpClient;
-    private readonly ICryptography cryptography;
+    private readonly ICryptographyFactory stateCryptography;
 
     public SignaturGruppen(
         ILogger<SignaturGruppen> logger,
         IOptions<AuthOptions> authOptions,
-        HttpClient httpClient, ICryptography cryptography
-    )
+        HttpClient httpClient,
+        ICryptographyFactory cryptography
+        )
     {
         this.logger = logger;
         this.authOptions = authOptions.Value;
         this.httpClient = httpClient;
-        this.cryptography = cryptography;
+        stateCryptography = cryptography;
```

Comment:
this could be `stateCryptography = cryptography.StateFoobar` and set the return type to `ICryptography`

Reply:
If we decide to keep it as a factory i think the name should be `cryptographyFactory` and not `stateCryptography`